hyperspy/drawing/utils.py

```diff
@@ -1035,9 +1040,20 @@ def transparent_single_color_cmap(color):
     if overlay:
         # Check if images all have same scale and therefore can be overlayed.
         for im in images:
-            if im.axes_manager[0].scale != images[0].axes_manager[0].scale:
+            axes = im.axes_manager.signal_axes
+            if axes[0].units != axes[1].units:
+                raise ValueError(
+                    "Images do not have the same units and should not be overlayed. "
+                    "Try converting units to the same value using "
+                    "`axes_manager.convert_units`."
+                )
+            # relative difference normalized to the size of the second axis
+            # the more pixels is in the second axis, the higher the tolerance need to be
```

Comment:
```suggestion
            # the more pixels are in the second axis, the higher the tolerance needs to be
```

as in #414 - "higher tolerance"?